### PR TITLE
Feature/survey close

### LINKED ISSUE #456: [FEATURE] Close survey on specific date
**Is your feature request related to a problem? Please describe.**
As a user I want to close a survey on a specfic date

**Describe the solution you'd like**
Close survey on specific date.

**Tasks**
Add "Close survey on date" toggle to Response Options. Use shadcn date picker to pick the date of closure.

Build an endpoint which is triggered by a CRON to close all surveys which are limited to a specific date.

We don't have a CRON yet but will build it soon :)

### How we code at Formbricks 🤓
- Everything is type-safe
- All UI components are in the package `formbricks/ui`
- Run `pnpm dev` to find a demo app to test in-app surveys at `localhost:3002`
- We use **chatGPT** to help refactor code. Use our [Formbricks ✨ megaprompt ✨](https://github.com/formbricks/formbricks/blob/main/megaprompt.md) to create the right context before you write your prompt.

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
1. Added Date picker in UI to select a `closeOnDate`
2. Added a POST endpoint to mark all the past surveys as completed(when being called from CRON)

Fixes #456 

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

## Checklist

- [ ] Added a screen recording or screenshots to this PR
- [ ] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx [hunk 1]
```diff
@@ -112,33 +156,79 @@ export default function ResponseOptionsCard({ localSurvey, setLocalSurvey }: Res
...
-          )}
+            )}
+          </div>
         </div>
       </Collapsible.CollapsibleContent>
     </Collapsible.Root>
   );
 }
+
+export function DatePicker({ date, handleDateChange }) {
```
Comment (jobenjada) [inline apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx:213]: Please move the datepicker to a separate, reusable component :)

File: packages/ui/components/Calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import * as React from "react";
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import { DayPicker } from "react-day-picker";
```
Comment (jobenjada) [inline packages/ui/components/Calendar.tsx:5]: Why did you decide to use React Day Picker instead of shadcn day picker?

**Review by jobenjada — CHANGES_REQUESTED** [review]
Overall it looks good, great work Piyush :)

When I double-click on a date, I get this error:

grafik

In terms of good UX, I suggest to **close the date picker, when a date is selected.**

File: apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx (hunk 1 above)
Comment (gupta-piyush19) [reply apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx:213]: sure, will do this.

File: packages/ui/components/Calendar.tsx (hunk 2 above)
Comment (gupta-piyush19) [reply packages/ui/components/Calendar.tsx:5]: shadcn date picker is using react-day-picker internally.

File: apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx (hunk 1 above)
Comment (jobenjada) [reply apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx]: thanks!

**Review by jobenjada — APPROVED** [review]

**Review by mattinannt — APPROVED** [review]
@gupta-piyush19 Great job 😊💪 Thanks a lot for this feature PR 🎉
I only added the closeOnDate field also to the new survey service in `packages/lib/services` as well as the new type definitions in `packages/types/v1`.

I will merge it into main now 😊

**Pull request merged**